jme3-core/src/main/java/com/jme3/input/FlyByCamera.java

```diff
@@ -382,28 +382,7 @@ protected void rotateCamera(float value, Vector3f axis){
      * @param value zoom amount
      */
     protected void zoomCamera(float value){
-        // derive fovY value
-        float h = cam.getFrustumTop();
-        float w = cam.getFrustumRight();
-        float aspect = w / h;
-
-        float near = cam.getFrustumNear();
-
-        float fovY = FastMath.atan(h / near)
-                / (FastMath.DEG_TO_RAD * .5f);
-        float newFovY = fovY + value * 0.1f * zoomSpeed;
-        if (newFovY > 0f) {
-            // Don't let the FOV go zero or negative.
-            fovY = newFovY;
-        }
-
-        h = FastMath.tan( fovY * FastMath.DEG_TO_RAD * .5f) * near;
-        w = h * aspect;
-
-        cam.setFrustumTop(h);
-        cam.setFrustumBottom(-h);
-        cam.setFrustumLeft(-w);
-        cam.setFrustumRight(w);
+        cam.setFov(cam.getFov() + value * 0.1F * zoomSpeed);

```

Comment:
It seems like a test is needed to prevent the FOV from going zero or negative. Currently, this code would throw an exception in that case. The old code never tried to set a zero or negative FOV.